WS17_18/01_introduction/your assignment/Yatodoa/Yatodoa/ViewController.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  ViewController.swift
+//  Yatodoa
+//
+//  Created by Uwe Müsse on 24.10.17.
+//  Copyright © 2017 Uwe Müsse. All rights reserved.
+//
+
+import UIKit
+
+class ViewController: UIViewController {
+    
+    @IBOutlet weak var solved: UILabel!
+    var todos: [String] = [] {
+        didSet {
+            let count = todos.count
+            solved.text = (count == 1) ? "\(count) Aufgabe erledigt" : "\(count) Aufgaben erledigt"
+        }
+    }
+
+    @IBOutlet weak var todoTextfield: UITextField!
+    @IBOutlet weak var todoTextfield2: UITextField!
+    
+    @IBAction func checkboxTapped(_ sender: UIButton) {
+        
+        checkboxHandling(sender, todoTextfield);
+        
+    }
+    @IBAction func checkbox2Tapped(_ sender: UIButton) {
+        
+        checkboxHandling(sender, todoTextfield2);
+        
+    }
+    
+    private func checkboxHandling (_ sender: UIButton, _ todoTextfield: UITextField){
+        
+        if let text = todoTextfield.text, !text.isEmpty{
+            if sender.currentTitle == nil {
+                sender.setTitle("✔", for: UIControlState.normal)
+                todoTextfield.alpha = 0.5
+                todoTextfield.isUserInteractionEnabled = false
+                todoTextfield.attributedText = NSAttributedString(string: text, attributes: [NSAttributedStringKey.strikethroughStyle: NSUnderlineStyle.styleSingle.rawValue])
```

Comment:
Ich habe in meiner Lösung zusätzlich die Schriftfarbe auf `UIColor.lightGray` gesetzt, sodass ich den `.alpha = 0.5` nicht mehr brauche. Die Farbe kann man auch über NSAttributedString setzen